src/components/Screen.tsx

```diff
@@ -165,6 +166,7 @@ export const InnerScreen = React.forwardRef<View, ScreenProps>(
   function InnerScreen(props, ref) {
     const innerRef = React.useRef<ViewConfig | null>(null);
     React.useImperativeHandle(ref, () => innerRef.current!, []);
+    const prevActivityState = usePrevious(props.activityState);
```

Comment:
I don't like running another useEffect every render, however this should be ok for now, since we want to relax this constraint in future.